hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/SplitPlanFirstComparator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.normalizer;
+
+import java.util.Comparator;
+
+/**
+ * Comparator class that gives higher priority to {@link SplitNormalizationPlan}.
+ */
+class SplitPlanFirstComparator implements Comparator<NormalizationPlan> {
```

Comment:
Why. And should we sort by table order too?

Reply:
Good question. I don't know the history of this class.

My original theory is something to do with splits vs. merges. If somehow a region is involved in both a split and a merge plan, the split would happen first and the merge would fail. This gives priority to splits (which I think is intentional), but is a bit ugly. I think a final validation and optimization pass should be taken over the resulting plan list, which is a superset of what's proposed in [HBASE-24418](https://issues.apache.org/jira/browse/HBASE-24418).

As for table ordering, the normalizer is invoked once per table, so there's no case where plans for multiple tables would be in the same collection. At least, not in current implementation.

Reply:
The PlanComparator compares plans based on if it is a split or merge. It puts all split plan at the head of the list and merge plans put to the end of the list. 
Based on the addToList order, split plans are already added first, followed by merge plans, so my understanding is that this sort does nothing. 
    plans.sort(planComparator);

We can get rid of this comparator totally and simplify the code. Something missing in my understanding?

Reply:
Reading through the old patches, i think this is an appendix we can safely discard.